src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -459,29 +456,25 @@ public static Image getGameIcon(final Window frame) {
     return img;
   }
 
-  public static void hostGame(final int port, final String playerName, final String comments, final String password,
+  public static void hostGame(
+      final int port,
+      final String playerName,
+      final String comments,
+      final String password,
       final Messengers messengers) {
-    final List<String> commands = new ArrayList<>();
-    ProcessRunnerUtil.populateBasicJavaArgs(commands);
-    commands.add("-D" + TRIPLEA_SERVER_PROPERTY + "=true");
-    commands.add("-D" + TRIPLEA_PORT_PROPERTY + "=" + port);
-    commands.add("-D" + TRIPLEA_NAME_PROPERTY + "=" + playerName);
-    commands.add("-D" + LOBBY_HOST + "="
-        + messengers.getMessenger().getRemoteServerSocketAddress().getAddress().getHostAddress());
-    commands.add("-D" + GameRunner.TRIPLEA_LOBBY_PORT_PROPERTY + "="
-        + messengers.getMessenger().getRemoteServerSocketAddress().getPort());
-    commands.add("-D" + LOBBY_GAME_COMMENTS + "=" + comments);
-    commands.add("-D" + LOBBY_GAME_HOSTED_BY + "=" + messengers.getMessenger().getLocalNode().getName());
+    System.setProperty(TRIPLEA_SERVER_PROPERTY, String.valueOf(true));
```

Comment:
Hmm, this seems like it could be a problem.  Do we have to revert the changes to these system properties when what was formerly the "spawned" process ends (e.g. when returning from a hosted game to the main menu)?  It just smells a bit to be changing (possibly multiple times) what are effectively global variables to control the execution path.

Reply:
Yes, but those changes don't affect the original process that spawned the new process.  Each JVM process gets its own copy of system properties.